src/passes/userDefinedTypesConverter.ts

```diff
@@ -76,15 +68,17 @@ export class UserDefinedTypesConverter extends ASTMapper {
   }
 
   visitFunctionCall(node: FunctionCall, ast: AST): void {
-    this.commonVisit(node, ast);
-    if (!(node.vExpression instanceof MemberAccess)) return;
-    if (!['unwrap', 'wrap'].includes(node.vExpression.memberName)) return;
-    const typeNode = getNodeType(node.vExpression.vExpression, ast.compilerVersion);
-    if (!(typeNode instanceof TypeNameType)) return;
-    if (!(typeNode.type instanceof UserDefinedType)) return;
-    if (!(typeNode.type.definition instanceof UserDefinedValueTypeDefinition)) return;
+    if (node.vExpression instanceof MemberAccess) {
+      if (['unwrap', 'wrap'].includes(node.vExpression.memberName)) {
+        const typeNode = getNodeType(node.vExpression.vExpression, ast.compilerVersion);
+        this.commonVisit(node, ast);
+        if (!(typeNode instanceof TypeNameType)) return;
+        if (!(typeNode.type instanceof UserDefinedType)) return;
+        if (!(typeNode.type.definition instanceof UserDefinedValueTypeDefinition)) return;
 
-    ast.replaceNode(node, node.vArguments[0]);
+        ast.replaceNode(node, node.vArguments[0]);
+      } else this.commonVisit(node, ast);
```

Comment:
I believe these should be visitExpression, not commonVisit so that if the typestring of the functionCall contains userdefinedvaluetypes then it can get corrected. Please correct me if I'm wrong

Reply:
Actually how I've designed this function is that, if the function call's expression is a member access of a user defined type, it replaces the `FunctionCall` node itself with it's argument expression. Since this function recurses bottom up, the argument expression would have already had it's typestring corrected when it replaces the `FunctionCall` node. 

But this is only to be done in the in the specific case when it's a `wrap` or `unwrap` function call from a user defined type. Otherwise it should recurse normally. Hence the two `else this.commonVisit(node, ast);` functions.

Also, when it is the case of a user defined type's member access, it's irrelevant weather we change the FunctionCall typestring or not since it's gonna get repalced anyway. But I capture the `typeNode` before calling the recursion otherwise the subsequent checks happen after the typestring is replaced and we never reach the `ast.replaceNode` call.

Please let me know if you think this approach is okay, or if I can make it better. Thanks!